File: 48-rotate-image/rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        for (int i = 0; i < matrix.length; i++) {

            for (int j = 0; j < matrix[i].length; j++) {

                if (i < j) {
                    int temp = matrix[i][j];
                    matrix[i][j] = matrix[j][i];
                    matrix[j][i] = temp;
                }

            }

        }

        for (int[] i : matrix) {
            int a = 0;
            int b = i.length - 1;

            while (a < b) {
                int temp = i[a];
                i[a] = i[b];
                i[b] = temp;
                a++;
                b--;
            }

        }
    }
}

File: 48-rotate-image/rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int row = matrix.size();
        int col = matrix[0].size();
        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                if (i == j)
                    break;

                swap(matrix[i][j], matrix[j][i]);
            }
        }

        for (int i = 0; i < row; i++) {
            reverse(matrix[i].begin(), matrix[i].end());
        }
    }
};